convert.py

```diff
@@ -104,6 +115,28 @@ def GenerateAudioContentForText(text):
             break
     
 
+def eupub_to_chapters():
+    path = os.path.abspath(EPUB_FILEPATH)
+    print(path)
+    book = epub.read_epub(path)
+
+    full_text = ''
+    for text in book.get_items_of_type(ebooklib.ITEM_DOCUMENT):
+        html_content = text.get_content().decode("utf-8")
+        full_text += html_content

```

Comment:
This is a famously and deceptive expensive operation. Since strings are immutable in Python, every single time you "append" content to the string, it doesn't actually modify the existing string - it actually performs a deep copy, and returns a reference to the new variable with all the data.

For more reference:
https://stackoverflow.com/questions/53236784/efficient-string-concatenation-in-python-3

Since we don't have a guaranteed ceiling on the size of the input file or its structure (number of ITEM_DOCUMENTS), this could actually get pretty expensive. 

Solution would be to do something like:

```
chapter_texts = []
for text in book.get_items_of_type(ebooklib.ITEM_DOCUMENT):
    chapter_texts.append(text.get_content().decode("utf-8"))
full_text = "".join(full_text)
```

Reply:
Good catch! This is really good to know generally - ty